provisioning_templates/snippet/kickstart_kernel_options.erb

```diff
@@ -0,0 +1,115 @@
+<%#
+kind: snippet
+name: kickstart_kernel_options
+model: ProvisioningTemplate
+snippet: true
+-%>
+<%
+  rhel_compatible = @host.operatingsystem.family == 'Redhat' && @host.operatingsystem.name != 'Fedora'
+  is_fedora = @host.operatingsystem.name == 'Fedora'
+  os_major = @host.operatingsystem.major.to_i
+  os_minor = @host.operatingsystem.minor.to_i
+  major = @host.operatingsystem.major.to_i
+  hostname = @host.name
+  iface = @host.provision_interface
+  mac = @host.provision_interface.mac
+  subnet4 = iface.subnet
+  subnet6 = iface.subnet6
+  options = []
+
+  if rhel_compatible && os_major < 8
+    # replaced with BOOTIF in EL8+
+    options.push("network", "ksdevice=bootif", "ks.device=bootif")
+  end
+
+  if mac
+    bootif = '00-' + mac.gsub(':', '-')
+    options.push("BOOTIF=#{bootif}")
+  end
+
+  # tell Anaconda what to pass off to kickstart server
+  # both current and legacy syntax provided
+  options.push("kssendmac", "ks.sendmac", "inst.ks.sendmac")
+
+  # networking credentials
+  raise("Dual-stack provisioning not supported") if subnet4 && subnet6
+  if subnet4 && subnet4.dhcp_boot_mode?
+    options.push("ip=dhcp") if rhel_compatible && major >= 7
+  elsif subnet4 && !subnet4.dhcp_boot_mode?
+    if rhel_compatible && major < 7
+      dns_servers = subnet4.dns_servers.join(',')
+      if @ipxe_net
+        options.push('ip=${netX/ip} netmask=${netX/netmask} gateway=${netX/gateway} dns=${dns}')
+      else
+        options.push("ip=#{iface.ip}", "netmask=#{subnet4.mask}", "gateway=#{subnet4.gateway}", "dns=#{dns_servers}")
+      end
+    else
+      if @ipxe_net
+        options.push("ip=${netX/ip}::${netX/gateway}:${netX/netmask}:#{hostname}:#{iface.identifier}:none nameserver=${dns}")
+      else
+        options.push("ip=#{iface.ip}::#{subnet4.gateway}:#{subnet4.mask}:#{hostname}:#{iface.identifier}:none")
```

Comment:
Aren't the nameservers missing here?

Reply:
iPXE has a special case, we actually want to add `nameserver=${dns}` there. But that could be overriden by DNS from subnet later on, I've ruled that out in the code:

```ruby
  if subnet4 && !@ipxe_net
    subnet4.dns_servers.each { |server|
      options.push("nameserver=#{server}")
    }
  elsif subnet6 && !@ipxe_net
    subnet6.dns_servers.each { |server|
      options.push("nameserver=#{server}")
    }
  end
```

In case of iPXE, these won't be appended.